srml/staking/src/lib.rs

```diff
@@ -1419,16 +1420,16 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+	fn on_session_ending(ending: SessionIndex, _: SessionIndex) -> Option<Vec<T::AccountId>> {
```

Comment:
This interface makes it seem like the next session index may not always be `current + 1`.

Reply:
```suggestion
	fn on_session_ending(_: SessionIndex, will_apply_at: SessionIndex) -> Option<Vec<T::AccountId>> {
```